txpool/txpool.go

```diff
@@ -247,13 +262,83 @@ func (p *TxPool) updatePending(i int64) {
 	metrics.SetGauge([]string{txPoolMetrics, "pending_transactions"}, float32(newPending))
 }
 
+func (p *TxPool) startGossipBatchers() {
+	for i := 0; i < batchersNum; i++ {
+		p.gossipWG.Add(1)
+
+		go p.gossipBatcher()
+	}
+}
+
+func (p *TxPool) stopGossipBatchers() {
+	close(p.gossipCh)
+	p.gossipWG.Wait()
+}
+
+func (p *TxPool) gossipBatcher() {
+	defer p.gossipWG.Done()
+
+	batch := make([]*types.Transaction, 0, p.gossipBatchSize)
+
+	tickerPeriod := time.Hour * 24 // reduce empty looping when no batching
+	if p.gossipBatchSize > 1 {
+		tickerPeriod = time.Millisecond * 500
+	}
+
+	ticker := time.NewTicker(tickerPeriod)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			if len(batch) > 0 {
+				p.publish(&batch)
+				batch = batch[:0]
+			}
+		case tx, chOpen := <-p.gossipCh:
+			if chOpen {
+				batch = append(batch, tx)
+				if len(batch) >= p.gossipBatchSize {
+					p.publish(&batch)
+					batch = batch[:0]
+				}
+			} else {
+				// flush when closing
+				if len(batch) > 0 {
+					p.publish(&batch)
+				}
+
+				return
+			}
+		}
+	}
+}
+
+func (p *TxPool) publish(batch *[]*types.Transaction) {
```

Comment:
Why do we need a pointer to a slice here?
```suggestion
func (p *TxPool) publish(batch []*types.Transaction) {
```

Reply:
Because slice reset is done within that method after publishing.